src/main/client_context.cpp

```diff
@@ -357,6 +357,7 @@ ClientContext::CreatePreparedStatementInternal(ClientContextLock &lock, const st
 	profiler.StartPhase("physical_planner");
 	// now convert logical query plan into a physical query plan
 	PhysicalPlanGenerator physical_planner(*this);
+	physical_planner.delim_index = planner.binder->GenerateTableIndex();
```

Comment:
Is there any particular reason we need to instantiate this to a newly generated table index? Wouldn't it be easier to read if the delim indexes were separate from table indexes and we just had "delim_scan 1", "delim scan 2" etc, instead of "delim scan 17", "delim scan 18"?

Reply:
This part is completely optional and purely preference. If we do it like that, all `CTE` and `DELIM` indices are unique, otherwise they may not be. But it does not matter either way, really. I'm fine dropping that bit of code.

Reply:
I like that. I will change the PR accordingly.